zerver/lib/event_schema.py

```diff
@@ -1516,13 +1516,15 @@ def check_update_global_notifications(
     assert isinstance(setting, setting_type)
 
 
+# user_id field is null for embedded variant of update_message
 update_message_required_fields = [
     ("type", Equals("update_message")),
-    ("user_id", int),
+    ("user_id", OptionalType(int)),
```

Comment:
I may not be understanding this correctly, but I thought this was for the type checking. It allows for a `null` (or None) value to be present as well as an integer.

I thought the test for the presence of the field is below in the comparison between expected and actual fields. And since it's in the `required_fields` group, it's checked in all of the different variations on the event schema.

Reply:
Okay. I added assertions for the `user_id` property in the if statement for the `is_embedded_update_only` argument (in addition to the check for the `rendering_only` property match). Let me know if that's not along the lines of what you were thinking.